### PR TITLE
Support Folia

### PR DESCRIPTION
**Pull request by Euphillya**
This version breaks Spigot and Paper compatibility before 1.20.1

Features :

- Support Folia (in progress)

Fix :

- [x] mcMMO https://github.com/Crazy-Crew/CrazyEnchantments/pull/809/commits/80cff79c7df0fbc655172fc7227a4bae85769d09 https://github.com/Crazy-Crew/CrazyEnchantments/pull/809/commits/563704b7b9f1a0a524fc71cd29768b87b73d046f
- [x] Metrics Support Folia

### CODE REVIEW

**Comment by Euphillya** [comment]
I don't think there is any other incompatibility.

File: paper/src/main/java/com/badbones69/crazyenchantments/paper/utils/Metrics.java [hunk 1]
```diff
@@ -0,0 +1,862 @@
...
+                this.value = value;
+            }
+
+            @Override
+            public String toString() {
+                return value;
+            }
+        }
+    }
+}
```
Comment (Euphillya) [inline paper/src/main/java/com/badbones69/crazyenchantments/paper/utils/Metrics.java:862]: I just need to know if we keep the Metrics or not.

File: paper/src/main/java/com/badbones69/crazyenchantments/paper/tasks/processors/ArmorMoveProcessor.java [hunk 2]
```diff
@@ -66,49 +66,51 @@ public void process(UUID playerId) {
...
-            if (player.getFoodLevel() < 20) {
-                checkImplants(armor, player, enchantments);
-            }
-
-            checkCommander(armor, player, enchantments);
-
-            if (SupportedPlugins.FACTIONS_UUID.isPluginLoaded()) {
-                final int radius = 4 + enchantments.get(CEnchantments.ANGEL.getEnchantment());
-                checkAngel(armor, player, enchantments, radius);
+        player.getScheduler().run(plugin, playerTask -> {
```
[the remark this answers is not in the document]
Comment (Euphillya) [reply paper/src/main/java/com/badbones69/crazyenchantments/paper/tasks/processors/ArmorMoveProcessor.java]: It was part of a code that I had done to test but which I mostly forgot to finish. I just updated.

File: paper/src/main/java/com/badbones69/crazyenchantments/paper/tasks/processors/ArmorMoveProcessor.java [hunk 3]
```diff
@@ -90,74 +91,75 @@ public void process(UUID playerId) {
             }
             useHellForge(player, armor, enchantments);
         }
+        player.getScheduler().run(this.plugin, task -> {
```
[the remark this answers is not in the document]
Comment (Euphillya) [reply paper/src/main/java/com/badbones69/crazyenchantments/paper/tasks/processors/ArmorMoveProcessor.java:94]: To access the inventory we must use the player scheduler

**Comment by Euphillya** [comment]
Is it good ?

**Comment by ryderbelserion** [comment]
You have unsigned commits blocking the merge.

**Comment by Euphillya** [comment]
Should I recreate this PR?

**Comment by ryderbelserion** [comment]
`git rebase --exec 'git commit --amend --no-edit -n -S' -i development`

This rebases everything till development (or any hash) and you don't have to copy paste after every commit.

https://superuser.com/questions/397149/can-you-gpg-sign-old-commits

**Comment by ryderbelserion** [comment]
> `git rebase --exec 'git commit --amend --no-edit -n -S' -i development` or use a commit hash like [6313ca1](https://github.com/Crazy-Crew/CrazyEnchantments/commit/6313ca1e7106006d917afd305ec5688ded8f611a)
> 
> This rebases everything till development (or any hash) and you don't have to copy paste after every commit.
> 
> https://superuser.com/questions/397149/can-you-gpg-sign-old-commits

or yeah you can re-create the branch and re-commit then open a new pull request

**Pull request closed without merging**